logbook-json/src/main/java/org/zalando/logbook/json/JacksonJsonFieldBodyFilter.java

```diff
@@ -0,0 +1,80 @@
+package org.zalando.logbook.json;
+
+import java.io.StringWriter;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import org.zalando.logbook.BodyFilter;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * 
+ * Thread-safe filter for JSON fields. Filters on property names, 
+ *
+ */
+
+public class JacksonJsonFieldBodyFilter implements BodyFilter {
+
+    private final String replacement;
+    private final Set<String> fields;
+    private final ObjectMapper objectMapper;
+    
+    public JacksonJsonFieldBodyFilter(Collection<String> fieldNames, String replacement, ObjectMapper objectMapper) {
+        this.fields = new HashSet<>(fieldNames); // thread safe for reading
+        this.replacement = replacement;
+        this.objectMapper = objectMapper;
+    }
+
+    public JacksonJsonFieldBodyFilter(Collection<String> fieldNames, String replacement) {
+        this(fieldNames, replacement, new ObjectMapper());
+    }
+
+    @Override
+    public String filter(String contentType, String body) {
+        return JsonMediaType.JSON.test(contentType) ? filter(body) : body;
+    }
+
+    public String filter(final String body) {
+        try {
+            JsonFactory factory = objectMapper.getFactory();
+            final JsonParser parser = factory.createParser(body);
+            
+            StringWriter writer = new StringWriter(body.length() * 2); // rough estimate of final size
+            
+            JsonGenerator generator = factory.createGenerator(writer);            
+            try {
+                while(true) {
+                    JsonToken nextToken = parser.nextToken();
+                    if(nextToken == null) {
+                        break;
+                    }
+
+                    generator.copyCurrentEvent(parser);
+                    if(nextToken == JsonToken.FIELD_NAME && fields.contains(parser.getCurrentName())) {
+                        nextToken = parser.nextToken();
+                        generator.writeString(replacement);
+                        if(!nextToken.isScalarValue()) {
+                            parser.skipChildren(); // skip children
+                        }
+                    }
+                }                    
+            } finally {
+                parser.close();
+                
+                generator.close();
+            }
+            
+            return writer.toString();
+        } catch(Exception e) {
+            // ignore
```

Comment:
Should we log something?

https://github.com/zalando/logbook/blob/f6321b4fcf3a726bb0135c6a1945350dcad9db2d/logbook-json/src/main/java/org/zalando/logbook/json/CompactingJsonBodyFilter.java#L40

Reply:
If this code defers logging to upstream, thats perhaps better? If the document is invalid, surely that will be caught during databinding.

Reply:
What kind of exception are we potentially ignoring here right now?

Reply:
Invalid JSON (or content classified as JSON but in another format somehow). Should not happen very often.

Reply:
Still, if it does seeing a log message would be nice.

Reply:
Added log message.